admin/installer/_images.py

```diff
@@ -0,0 +1,83 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Helper utilities for CloudFormation Installer's Packer images.
+"""
+from pyrsistent import freeze
+
+
+class _PackerOutputParser(object):
+    """
+    Parse the output of ``packer -machine-readable``.
+    """
+    def __init__(self):
+        self.artifacts = []
+        self._current_artifact = {}
+
+    def _parse_line_ARTIFACT(self, parts):
+        """
+        Parse line parts containing information about an artifact.
+
+        :param list parts: The parts of resulting from splitting a comma
+            separated packer output line.
+        """
+        artifact_type = parts[1]
+        if parts[4] == 'end':
+            self._current_artifact['type'] = artifact_type
+            self.artifacts.append(self._current_artifact)
+            self._current_artifact = {}
```

Comment:
nit: also return here, or put the rest in an else block?

Otherwise you are adding an {'end' : []} key to the end of the next _current_artifact.